kratos/processes/from_json_check_result_process.h

```diff
@@ -0,0 +1,921 @@
+        }
+    }
+
+//     /**
+//      * @brief This method retrieves the interpolated value from the database
+//      * @param rVariable The variable to be retrieved
+//      * @param EntityIndex The index of the entity (not the entity Id, the index in the database)
+//      * @param Time The time value to be retrieved
+//      * @param ComponentIndex The component index of the vector/array
+//      * @param GPIndex The Gauss point index
+//      * @tparam TVariableType The variable type considered
+//      */
+//     template<class TVariableType>
+//     const double GetValue(
+//         const TVariableType& rVariable,
+//         const IndexType EntityIndex,
+//         const double Time,
+//         const SizeType ComponentIndex = 0,
+//         const SizeType GPIndex = 0
+//         ) const
+//     {
+//         return GetVariableData(rVariable).GetValue(EntityIndex, Time, ComponentIndex, GPIndex);
+//     }
+
+    /**
+     * @brief This method set the values into the tables
+     * @param rValuesX The values of the X axis
+     * @param rValuesY The values of the Y axis
+     * @param rVariable The variable to be retrieved
+     * @param EntityIndex The index of the entity (not the entity Id, the index in the database)
+     * @param ComponentIndex The component index of the vector/array
+     * @param GPIndex The GP index
+     * @tparam TVariableType The variable type considered
+     */
+    template<class TVariableType>
+    void SetValues(
+        const Vector& rValuesX,
+        const Vector& rValuesY,
+        const TVariableType& rVariable,
+        const IndexType EntityIndex,
+        const SizeType ComponentIndex = 0,
+        const SizeType GPIndex = 0
+        )
+    {
+        auto it = find(rVariable.Key());
+        if (it != end()) {
+            auto& r_database = it->second;
+            r_database.SetValues(rValuesX, rValuesY, EntityIndex, ComponentIndex, GPIndex);
+        } else {
+            KRATOS_ERROR << "Not allocated Variable: " << rVariable.Name() << std::endl;
+        }
+    }
+
+    /**
+     * @brief This method clears the database of the common column vector
+     */
+    void ClearCommonColumn()
+    {
+        mCommonColumn.clear();
+    }
+
+    /**
+     * @brief This function is designed for clear all the databases
+     */
+    void Clear()
+    {
+        this->clear();
+        ClearCommonColumn();
+    }
+
+    /**
+     * @brief This function is designed for being called after ExecuteInitialize ONCE to verify that the input is correct.
+     */
+    int Check()
+    {
+        // Doing check in the table size
+        for (auto& r_pair : *this) {
+            auto& r_table_vector_vector_vector = r_pair.second;
+            for (auto& r_table_vector_vector : r_table_vector_vector_vector) {
+                for (auto& r_table_vector : r_table_vector_vector) {
+                    for (auto& p_table : r_table_vector) {
```

Comment:
Have you tried to do these `const auto&`? (I'm not sure if the `const` version of the methods below is available).